tests/ballerina-unit-test/src/test/resources/logging.properties

```diff
@@ -22,7 +22,7 @@ java.util.logging.ConsoleHandler.level=ALL
 java.util.logging.ConsoleHandler.formatter=org.ballerinalang.logging.formatters.BallerinaLogFormatter
 org.ballerinalang.logging.formatters.BallerinaLogFormatter.format=%1$tY-%1$tm-%1$td %1$tH:%1$tM:%1$tS,%1$tL %2$s [%3$s] - %4$s %n
 
-org.ballerinalang.test.utils.TestLogHandler.level=INFO
+org.ballerinalang.test.utils.TestLogHandler.level=FINER
```

Comment:
This cause to print all Debug logs while building the unit tests. Are we testing any debug log specific test case ? Please revert this change.

Reply:
Yes, this is purposefully added. Debug available only for the IO package. The rest will have their default behavior.

Reply:
what is the purpose of having debug log for IO package?

Reply:
There are some meta level information maintain in the channels and sockets. These things will not interest to the end user and they don't use it. But native developers need this information to verify some information. Also make sure to debug logs give necessary information. We don't have any other mechanism to test whether debug working unless we test that manually.

Reply:
Ok, I have asked a wrong question. You can have debug logs in your package. I meant to ask was, what is the purpose of enabling debug logs only for IO pkg in test cases. ? 

Manually testing is not a practical and enabling debug logs in unit tests not going to test it either. Write a proper test case then.

Reply:
I asked [1] the current following practices when it comes to log in debug mode when it come to ballerina native development. Based on an outcome of that discussion, we can either remove or keep the debug log. 

Enabling debug in the test will give us guarantees of the execution of the log without any issue and we can further improve the test by verifying the log whether we are getting expected outcome.

[1] - https://ballerina-platform.slack.com/archives/C9WMA6QDC/p1535002330000100

Reply:
Ok, I understood your intention now. you want to improve the code coverage by enabling debug logs. But will it test the functionality?

Reply:
I still don't buy your point. If we enable debug and that causes NPE means bad coding. The developer has to check for possible null pointers, even it is a tracing code.

Still, you don't need to change global log level. Setting log level for your classes will be sufficient to invoke hit your code.

Reply:
When the developer does the initial implementation, developer know where it can cause the issues. So there can be certain areas that aren't giving any issues at all that time. They can use this information for debug purpose as well. But later due to changes in some other areas this can affect. So based on your logic, we have to do the null check for every variable in every line which is not scalable. Which exactly happen in the past as well.

This will not change the global log level. Apart from this you need to add the package again with the necessary log level. Just simply changing this won't active debug globally.

Reply:
Why checking null is not scalable? This is a serious statement. Please schedule a code review for Sockets implementation next week.